File: src/main/java/net/reflact/client/hud/overlay/StatusEffectOverlay.java
package net.reflact.client.hud.overlay;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.gui.DrawContext;
import net.reflact.client.ReflactClient;
import java.util.Collection;
import net.minecraft.entity.effect.StatusEffectInstance;

public class StatusEffectOverlay implements ReflactOverlay {
    @Override
    public void render(DrawContext context, float tickDelta) {
        MinecraftClient client = MinecraftClient.getInstance();
        if (client.player == null) return;
        int x = getX();
        int y = getY();
        int effectOffset = 0;
        Collection<StatusEffectInstance> effects = client.player.getStatusEffects();
        if (!effects.isEmpty()) {
             for (StatusEffectInstance effect : effects) {
                 net.minecraft.text.Text name = effect.getEffectType().value().getName();
                 String duration = net.minecraft.util.StringHelper.formatTicks(effect.getDuration(), client.world.getTickManager().getTickRate());
                 String amplifier = effect.getAmplifier() > 0 ? " " + (effect.getAmplifier() + 1) : "";
                 String text = name.getString() + amplifier + " (" + duration + ")";
                 int color = effect.getEffectType().value().isBeneficial() ? 0xFF55FF55 : 0xFFFF5555;
                 context.drawText(client.textRenderer, text, x, y + effectOffset, color, true);
                 effectOffset += 10;
             }
        }
    }
    @Override public int getX() { return ReflactClient.CONFIG.statusEffectHudX(); }
    @Override public int getY() { return ReflactClient.CONFIG.statusEffectHudY(); }
    @Override public void setX(int x) { ReflactClient.CONFIG.statusEffectHudX(x); }
    @Override public void setY(int y) { ReflactClient.CONFIG.statusEffectHudY(y); }
    @Override public int getWidth() { return 100; }
    @Override public int getHeight() { return 100; } // Dynamic
    @Override public boolean isEnabled() { return ReflactClient.CONFIG.showStatusEffectHud(); }
    @Override public String getName() { return "Status Effects"; }
}

File: src/main/kotlin/net/reflact/client/hud/overlay/StatusEffectOverlay.kt
package net.reflact.client.hud.overlay

import net.minecraft.client.MinecraftClient
import net.minecraft.client.gui.DrawContext
import net.minecraft.util.StringHelper
import net.reflact.client.ReflactClient

class StatusEffectOverlay : ReflactOverlay {
    override fun render(context: DrawContext, tickDelta: Float) {
        val client = MinecraftClient.getInstance()
        if (client.player == null) return
        val x = x
        val y = y
        var effectOffset = 0
        val effects = client.player!!.statusEffects
        if (effects.isNotEmpty()) {
            for (effect in effects) {
                val name = effect.effectType.value().name
                val duration = StringHelper.formatTicks(effect.duration, client.world!!.tickManager.tickRate)
                val amplifier = if (effect.amplifier > 0) " " + (effect.amplifier + 1) else ""
                val text = name.string + amplifier + " (" + duration + ")"
                val color = if (effect.effectType.value().isBeneficial) 0xFF55FF55.toInt() else 0xFFFF5555.toInt()
                context.drawText(client.textRenderer, text, x, y + effectOffset, color, true)
                effectOffset += 10
            }
        }
    }

    override var x: Int
        get() = ReflactClient.CONFIG.statusEffectHudX()
        set(value) { ReflactClient.CONFIG.statusEffectHudX(value) }
    override var y: Int
        get() = ReflactClient.CONFIG.statusEffectHudY()
        set(value) { ReflactClient.CONFIG.statusEffectHudY(value) }
    override var width: Int get() = 100; set(value) {}
    override var height: Int get() = 100; set(value) {}
    override val isEnabled: Boolean get() = ReflactClient.CONFIG.showStatusEffectHud()
    override val name: String get() = "Status Effects"
}
